backend/bot/index.js

```diff
@@ -36,12 +68,58 @@ const postResultToSlack = (match) => {
     prefix = 'HOLY SHIT! ';
   }
 
-  const suffix = isComedyDuo?  ':marioluigi:' : '';
+  const suffix = isComedyDuo ? ':marioluigi:' : '';
 
   const messageText = `${prefix}${winningPlayers.join(', ')} just beat ${losingPlayers.join(', ')}. Rating change: ${ratingChange}${suffix}`
-  bot.postMessage(channel.id, messageText, {as_user: true});
+  await bot.postMessage(channel.id, messageText, { as_user: true });
 }
 
+const postRankingChangeMessage = async (oldRankings, newRankings) => {
+  const rankingChanges = []
+  oldRankings.forEach((oldPlayer, index) => {
```

Comment:
Please use map with index argument instead of pushing to a mutable array.

Reply:
How would this work if I'm only pushing when `newPlayer.id !== oldPlayer.id`? 🤔

Should I just `map` it and then use `filter`?

Reply:
or like this?
```
const hasLeaderboardChanged = (leaderboardSize, oldRankings, newRankings) => (
  oldRankings.findIndex((oldPlayer, index) => oldPlayer.id !== newRankings[index].id) < leaderboardSize
```

Reply:
You can do it like this:
```
const rankingChanges = oldRankings.map((oldPlayer, index) => ({
  name: oldPlayer.name,
  oldRanking: index + 1,
  newRanking: newRankings.findIndex(p => p.id === oldPlayer.id) + 1
})).filter(ranking => ranking.oldRanking != ranking.newRanking)
```

Reply:
Yeah that's what I meant, I just wasn't sure if it's what you meant too

Reply:
Actually I think this way there may be a problem if we had several people with the same rating, since we are only filtering by comparing ratings.

Reply:
Ranking is the order - you can't have 2 ppl with the same one :D

Reply:
Exactly, I deleted my previous comment immediately after posting it because I realized that.